py/server/deephaven_internal/stream.py

```diff
@@ -0,0 +1,87 @@
+import io
+
+
+class TeeStream(io.TextIOBase):
+    """TextIOBase subclass that splits output between a delegate instance and a set of lambdas.
+
+    Can delegate some calls to the provided stream but actually write only to the lambdas. Useful to adapt any existing
+    file-like object (such as sys.stdout or sys.stderr), but actually let some other processing take place before
+    writing.
+    """
+
+    @classmethod
+    def split(cls, py_stream, java_stream):
+        return TeeStream(
+            orig_stream=py_stream,
+            should_write_to_orig_stream=True,
+            write_func=lambda t: java_stream.write(bytes(t, py_stream.encoding)),
+            flush_func=lambda: java_stream.flush(),
+            close_func=lambda: java_stream.close()
+        )
+
+    @classmethod
+    def redirect(cls, py_stream, java_stream):
+        if hasattr(py_stream, "encoding"):
+            encoding = py_stream.encoding
+        else:
+            encoding = 'UTF-8'
+        return TeeStream(
+            orig_stream=py_stream,
+            should_write_to_orig_stream=False,
+            write_func=lambda t: java_stream.write(bytes(t, encoding)),
+            flush_func=lambda: java_stream.flush(),
+            close_func=lambda: java_stream.close()
```

Comment:
I'm wondering - would it make sense to "simplify" and remove the lambdas?

```python
TeeStream(
  orig_stream=py_stream,
  other_stream=java_stream,
  should_write_to_orig_stream=...)
```

The write implementation could check it's own `encoding` property as you have it written.

Reply:
Possibly, I was imagining that while my current use case was limited to redirecting into or splitting into the JVM, we might case about generally teeing or redirecting output at a later date, and the io.IOBase type is not fun to work with - once I had the specifics worked out, I have a constructor that is easy to generalize, and two factory methods that are specific to their java use cases.